### PR TITLE
special parameter errors

### PR DESCRIPTION
**Pull request by njr-11**
Covers error paths involving special parameters, including:
* duplicate and intermixed PageRequest/Limit
* incorrectly ordered special parameters

- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Review by jmstephensgit — DISMISSED** [review]

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f3884b99-1ec7-4d63-be8a-465bdb961b88) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

File: dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops [hunk 1]
```diff
@@ -970,3 +970,13 @@ CWWKD1097.param.incompat=CWWKD1097E: The {0} method of the {1} repository \
...
  not compatible with the type of repository operation.
 CWWKD1097.param.incompat.useraction=Remove the Order, Sort, or Limit parameter.
+
+CWWKD1098.spec.param.position.err=CWWKD1098E: The {0} method of the {1} repository \
+ has a {2} parameter that is positioned before the query parameters. \
+ Parameter types that have special meaning in Jakarta Data ({3}) must be \
+ positioned after the other parameters in the repository method signature.
+CWWKD1098.spec.param.position.err.explanation=When special Jakarta Data defined \
+ parameter types are included in a repository method signture, they must be \
+ positioned after all of the query parameters.
```
Comment (helyarp) [inline dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops]: Change "all of the" to "all the" or "all" (remove "of").

I use "of" when I talk, but the "of" gets flagged by a content quality checker.

**Review by helyarp — COMMENTED** [review]
Suggested one small change, otherwise the messages are clear

File: dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops (hunk 1 above)
Comment (njr-11) [reply dev/io.openliberty.data.internal.persistence/resources/io/openliberty/data/internal/persistence/resources/CWWKDMessages.nlsprops]: Thanks for pointing that out. I added commit https://github.com/OpenLiberty/open-liberty/pull/30405/commits/2208ef554f19120a714f75fefaeb220a1ea23ed4 to correct it.

**Review by jmstephensgit — APPROVED** [review]

**Comment by njr-11** [comment]
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f3d9788a-e034-49f1-8992-5b86c0ea1e18) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

**Review by helyarp — COMMENTED** [review]
Looks good!

**Pull request merged**